### PR TITLE
Fix: SAM Crashes Invalid swagger models exception

### PR DESCRIPTION
**Pull request by prenx4x**
*Issue #, if available:*

*Description of changes:*

*Description of how you validated changes:*

*Checklist:*

- [x] Write/update tests
- [x] `make pr` passes
- [ ] Update documentation
- [ ] Verify transformed template deploys and application functions as expected

*Examples?*

Please reach out in the comments, if you want to add an example. Examples will be 
added to `sam init` through https://github.com/awslabs/aws-sam-cli-app-templates/

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: samtranslator/swagger/swagger.py [hunk 1]
```diff
@@ -852,6 +852,8 @@ def add_resource_policy(self, resource_policy, path, api_id, stage):
         """
         if resource_policy is None:
             return
+        if not isinstance(resource_policy, dict):
+            raise InvalidDocumentException([InvalidTemplateException("Resource Policy is not a valid dictionary.")])
```
Comment (aahung) [inline samtranslator/swagger/swagger.py:856]: LGTM, this is the fix and others are just changing the exception types right?

**Review by aahung — APPROVED** [review]

File: samtranslator/swagger/swagger.py (hunk 1 above)
Comment (c2tarun) [inline samtranslator/swagger/swagger.py:856]: Curious, what is benefit of wrapping an array of exceptions in  another exception?

Comment (aahung) [reply samtranslator/swagger/swagger.py:856]: This is how it is meant to be used
```
    Attributes:
        message -- explanation of the error
        causes -- list of errors which caused this document to be invalid
    """
```

Comment (wchengru) [reply samtranslator/swagger/swagger.py:856]: +1 on this question. Also, not sure if resource_policy here is allowed to be an instrinsic expression, like `!Ref: sth`. If so, maybe we need to consider this.

**Review by c2tarun — APPROVED** [review]

File: samtranslator/swagger/swagger.py (hunk 1 above)
Comment (prenx4x) [reply samtranslator/swagger/swagger.py:856]: No idea why. This is what InvalidDocumentException needs. I saw other examples and this is what they did. I tried other ways but none works.

Comment (prenx4x) [reply samtranslator/swagger/swagger.py:856]: that and adding tests so to not regress in future.

Comment (prenx4x) [reply samtranslator/swagger/swagger.py:856]: afaict, InvalidDocumentException is meta exception, which counts all sub exceptions and then produces a single exception message combining them all.

Comment (prenx4x) [reply samtranslator/swagger/swagger.py:856]: @wchengru I am not sure sure what you mean. Can you explain further?

Comment (prenx4x) [reply samtranslator/swagger/swagger.py:856]: As per @wchengru 's suggestion tried a template with resourcepolicy: !ref but did not work. We probably do not support it here.

**Pull request merged**